Cubical/Categories/Category.agda

```diff
@@ -24,59 +24,90 @@ module Cubical.Categories.Category where
 open import Cubical.Core.Glue
 open import Cubical.Foundations.Prelude
 
+private
+  variable
+    ℓ ℓ' : Level
+
 -- Precategories
 
 record Precategory ℓ ℓ' : Type (ℓ-suc (ℓ-max ℓ ℓ')) where
   no-eta-equality
   field
     ob : Type ℓ
-    hom : ob → ob → Type ℓ'
-    idn : ∀ x → hom x x
-    seq : ∀ {x y z} (f : hom x y) (g : hom y z) → hom x z
-    seq-λ : ∀ {x y : ob} (f : hom x y) → seq (idn x) f ≡ f
-    seq-ρ : ∀ {x y} (f : hom x y) → seq f (idn y) ≡ f
-    seq-α : ∀ {u v w x} (f : hom u v) (g : hom v w) (h : hom w x) → seq (seq f g) h ≡ seq f (seq g h)
+    Hom[_,_] : ob → ob → Type ℓ'
+    id : ∀ x → Hom[ x , x ]
+    _⋆_ : ∀ {x y z} (f : Hom[ x , y ]) (g : Hom[ y , z ]) → Hom[ x , z ]
+    seq-λ : ∀ {x y : ob} (f : Hom[ x , y ]) → (id x) ⋆ f ≡ f
+    seq-ρ : ∀ {x y} (f : Hom[ x , y ]) → f ⋆ (id y) ≡ f
+    seq-α : ∀ {u v w x} (f : Hom[ u , v ]) (g : Hom[ v , w ]) (h : Hom[ w , x ]) → (f ⋆ g) ⋆ h ≡ f ⋆ (g ⋆ h)
+
+  -- composition: alternative to diagramatic order
+  _∘_ : ∀ {x y z} (g : Hom[ y , z ]) (f : Hom[ x , y ]) → Hom[ x , z ]
+  g ∘ f = f ⋆ g
 
 open Precategory public
 
+
+-- Helpful syntax/notation
+
+_[_,_] : (C : Precategory ℓ ℓ') → (x y : C .ob) → Type ℓ'
+_[_,_] = Hom[_,_]
+
+-- needed to define this in order to be able to make the subsequence syntax declaration
+seq' : ∀ (C : Precategory ℓ ℓ') {x y z} (f : C [ x , y ]) (g : C [ y , z ]) → C [ x , z ]
+seq' = _⋆_
+
+infix 15 seq'
+syntax seq' C f g = f ⋆⟨ C ⟩ g
+
+-- composition
+comp' : ∀ (C : Precategory ℓ ℓ') {x y z} (g : C [ y , z ]) (f : C [ x , y ]) → C [ x , z ]
+comp' = _∘_
+
+infix 16 comp'
+syntax comp' C g f = g ∘⟨ C ⟩ f
+
+
+
 -- Categories
 
-record isCategory {ℓ ℓ'} (𝒞 : Precategory ℓ ℓ') : Type (ℓ-max ℓ ℓ') where
+record isCategory (C : Precategory ℓ ℓ') : Type (ℓ-max ℓ ℓ') where
   field
-    homIsSet : ∀ {x y} → isSet (𝒞 .hom x y)
+    homIsSet : ∀ {x y} → isSet (C [ x , y ])
 
-open isCategory public
 
 -- Isomorphisms and paths in precategories
 
-record CatIso {ℓ ℓ' : Level} {𝒞 : Precategory ℓ ℓ'} (x y : 𝒞 .ob) : Type ℓ' where
+record CatIso {C : Precategory ℓ ℓ'} (x y : C .Precategory.ob) : Type ℓ' where
   constructor catiso
   field
-    h : 𝒞 .hom x y
-    h⁻¹ : 𝒞 .hom y x
-    sec : 𝒞 .seq h⁻¹ h ≡ 𝒞 .idn y
-    ret : 𝒞 .seq h h⁻¹ ≡ 𝒞 .idn x
+    h : C [ x , y ]
+    h⁻¹ : C [ y , x ]
+    sec : (h⁻¹ ⋆⟨ C ⟩ h) ≡ C .id y
```

Comment:
Why do you need parentheses here? Maybe something should bind differently?

Reply:
don't need parentheses